### DISCUSSION TITLE
Issue with NextAuth destroying the session?

### DISCUSSION THREAD

**Discussion by miguelguadarrama** [nextauthjs/next-auth#2370] [Help]
In my app I have a wrapper layout component over the whole application in the _app.js file which evaluates the useSession()'s loading value to determine if a <Loader /> should show or it can render the props.children. Which in theory works perfectly because it will just show a loader one time and once the session is ready (e.g. loading === false) it will display the page accordingly to the session state. And because of SPA, we never should see the loader again.

The problem is, for some reason NextAuth likes to destroy the session at random times. I use Okta provider, and the timeout is set to 8 hours for the OpenID tokens. When the session destroy happens, the loading value becomes true again so my wrapper layout component shows the Loader component again for a few seconds. What I'm trying to understand is why NextAuth is destroying the session, even though the OpenID tokens (access tokens) still have hours before becoming invalid.

I also noticed in the NextJS console output this line whenever this happens:

`{ loading: true, session: undefined }`
`event - build page: /api/auth/[...nextauth]`

Which, could be the culprit. However, this also happened a few times in my deployed app. So not sure if it's related to fast refresh. Anyway what could be triggering a rebuild of nextauth backend component?

Hopefully someone has had a similar issue or can provide some suggestions.

**Comment by balazsorban44** [comment]
The session validity is independent of your provider's token expiry. You could say `next-auth` doesn't really know what your provider returns, it is just regular data.

The session is considered expired when the session cookie is removed. By default, this is 30 days of inactivity. In case the `/api/auth/session`  (eg.: `useSession`, `getSession`) is invoked before expiry, that 30 days is reset. (In other words, we rotate the session expiry, it is not an absolute value).

You can override the default expiry with the `maxAge` option: https://next-auth.js.org/configuration/options#session

By default a session is never considered stale from the client's perspective, meaning it won't get a new session in the lifetime of your SPA (eg. a reload would re-fetch the session).

You can change that behaviour here:
https://next-auth.js.org/getting-started/client#options

You can also poll the session regularly in the background, to keep it active, see the above link for that too.

**Reply by miguelguadarrama** [reply]
Hi thanks for this, and sorry for getting back so late to you.

I think my specific case is not that it reaches some expiration date (e.g., 30 days). In my case this seems to happen at random, every 5 minutes or so (didn't actually measure it, just a guess). I can be browsing different sections of my SPA fine, and at some point the session (from useSession()) will return:

`{ loading: true, session: undefined }`

Which causes the app to enter the "loading" state (it blanks the whole page with a progress bar...) so it interrupts whatever the user was doing (e.g., loses the current component state), but returns quickly to the same component/page.

I will for now play around with the <Provider> options as you mentioned above, maybe some configuration can help prevent this.
